core/node/api_server/src/execution_sandbox/execute.rs

```diff
@@ -170,23 +170,34 @@ impl TransactionExecutor {
         vm_permit: VmPermit,
         shared_args: TxSharedArgs,
         connection_pool: ConnectionPool<Core>,
+        enforced_base_fee: Option<u64>,
         mut tx: L2Tx,
         block_args: BlockArgs,
         vm_execution_cache_misses_limit: Option<usize>,
         custom_tracers: Vec<ApiTracer>,
     ) -> anyhow::Result<VmExecutionResultAndLogs> {
-        let enforced_base_fee = tx.common_data.fee.max_fee_per_gas.as_u64();
         let execution_args =
             TxExecutionArgs::for_eth_call(enforced_base_fee, vm_execution_cache_misses_limit);
 
         if tx.common_data.signature.is_empty() {
             tx.common_data.signature = PackedEthSignature::default().serialize_packed().into();
         }
 
+        let mut connection = connection_pool
+            .connection_tagged("api")
+            .await
+            .context("failed acquiring DB connection")?;
+        let protocol_version = block_args
+            .resolve_block_info(&mut connection)
+            .await
+            .context("failed to resolve block info")?
+            .protocol_version;
+        drop(connection);
+
         // Protection against infinite-loop eth_calls and alike:
         // limiting the amount of gas the call can use.
         // We can't use `BLOCK_ERGS_LIMIT` here since the VM itself has some overhead.
-        tx.common_data.fee.gas_limit = ETH_CALL_GAS_LIMIT.into();
+        tx.common_data.fee.gas_limit = get_eth_call_gas_limit(protocol_version.into()).into();
```

Comment:
i'm not sure it's the correct place for setting it. Probably, it's better to pass an argument, `use_max_limit` or so